.github/workflows/backend-build.yml

```diff
@@ -0,0 +1,92 @@
+name: "* Build backend"
+on:
+    workflow_call:
+        inputs:
+            node-version:
+                required: true
+                type: string
+            upload-artifact:
+                description: Upload artifact to Github artifact repository
+                type: boolean
+                default: false
+            tag:
+                required: false
+                type: string
+        outputs:
+            version:
+                description: "The version built"
+                value: ${{ jobs.configure.outputs.version }}
+            artifact:
+                description: "The name of the artifact in Github repo"
+                value: ${{ jobs.configure.outputs.artifact }}
+            outcome:
+                description: "Outcome from upload"
+                value: ${{ jobs.build.outputs.outcome }}
+
+jobs:
+
+    configure:
+        runs-on: ubuntu-latest
+        defaults:
+            run:
+                working-directory: ./server/
+        outputs:
+            version: ${{ steps.vars.outputs.version }}
+            artifact: ${{ steps.vars.outputs.artifact }}
+
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@v3
+
+            - name: Fetch tag ${{ inputs.tag }}
+              if: inputs.tag != ''
+              run: |
+                  git fetch --tags
+                  git checkout ${{ inputs.tag }}
+
+            - name: Configure variables
+              id: vars
+              run: |
+                  version=$(cat package.json | jq -r '.version')
+                  echo "version=$version" >> $GITHUB_OUTPUT
+                  echo "artifact=mikane-dist-$version" >> $GITHUB_OUTPUT
```

Comment:
Isn't this the same name as the frontend artifact? Won't it be overwritten if run at the same time? Or it might download frontend artifact later if that was uploaded between backend build and deploy

Reply:
True - will fix